src/composables/multisigAccounts.ts

```diff
@@ -175,12 +176,109 @@ export function useMultisigAccounts({
     }
   }
 
+  function splitArrayIntoFives(arr: any) {
+    const result = [];
+    for (let i = 0; i < arr.length; i += 5) {
+      result.push(arr.slice(i, i + 5));
+    }
+    return result;
+  }
+
   /**
-   * Refresh the list of the multisig accounts.
+   * Get extended data for a multisig account
    */
-  async function updateMultisigAccounts() {
+  async function getMultisigAccountInfo(
+    contractId: Encoded.ContractAddress,
+    gaAccountId: string,
+    otherMultisigData: any,
+  ): Promise<IMultisigAccount> {
     const dryAeSdk = await getDryAeSdk();
+    try {
+      const contractInstance = await dryAeSdk.initializeContract({
+        aci: SimpleGAMultiSigAci,
+        address: contractId,
+      });
 
+      const currentAccount = multisigAccounts.value
+        .find((account) => account.contractId === contractId);
+
+      const [
+        nonce,
+        signers,
+        consensusResult,
+        balance,
+      ] = (await Promise.all([
+        (
+          (isAdditionalInfoNeeded.value && gaAccountId === activeMultisigAccountId.value)
+          || currentAccount?.nonce == null
+        )
+          ? contractInstance.get_nonce()
+          : { decodedResult: currentAccount.nonce },
+        currentAccount?.signers
+          ? { decodedResult: currentAccount.signers }
+          : contractInstance.get_signers(),
+        contractInstance.get_consensus_info(),
+        gaAccountId ? dryAeSdk.getBalance(gaAccountId as Encoded.AccountAddress) : 0,
+      ]));
+
+      const decodedConsensus = consensusResult.decodedResult;
+      const txHash = decodedConsensus.tx_hash as Uint8Array;
+      const consensus: IMultisigConsensus = camelCaseKeysDeep(decodedConsensus);
+
+      consensus.expirationHeight = Number(consensus.expirationHeight);
+      consensus.confirmationsRequired = Number(consensus.confirmationsRequired);
+
+      const hasPendingTransaction = !!txHash && !consensus.expired;
+
+      return {
+        ...consensus,
+        ...otherMultisigData,
+        contractId,
+        gaAccountId,
+        nonce: Number(nonce.decodedResult),
+        signers: signers.decodedResult,
+        balance: toShiftedBigNumber(balance, -AE_COIN_PRECISION),
+        hasPendingTransaction,
+        txHash: txHash ? Buffer.from(txHash).toString('hex') : undefined,
+      };
+    } catch (error) {
+      /**
+       * Node might throw nonce mismatch error, skip the current account update
+       * return the existing data and account details will be updated in the next poll.
+       */
+      if (!(error instanceof DryRunError)) {
+        handleUnknownError(error);
+      }
+      return multisigAccounts.value.find(
+        (account) => account.contractId === contractId,
+      ) as IMultisigAccount;
```

Comment:
I don't think the `as IMultisigAccount` is needed here. If you want to inform TS that you are sure that the result won't be undefined please use `!` at the end of line.